File: code/csharp/DesignPatterns/proxy/RealLegalService.cs
namespace Proxy;
using System;

class RealLegalService : ILegalService
{
    public void RequestLegalAssistance()
    {
        Console.WriteLine("Real legal service is providing legal assistance.");
    }

    public void ProvideLegalAssistance()
    {
        RequestLegalAssistance();
    }
}

File: code/typescript/src/proxy/RealLegalService.ts
import { LegalService } from "./LegalService";

export class RealLegalService implements LegalService {
    requestLegalAssistance(): void {
        console.log("Real legal service is providing legal assistance.");
    }

    provideLegalAssistance(): void {
        this.requestLegalAssistance();
    }
}
